packages/vite/src/node/logger.ts

```diff
@@ -78,7 +81,23 @@ export function createLogger(
     allowClearScreen && process.stdout.isTTY && !process.env.CI
   const clear = canClearScreen ? clearScreen : () => {}
 
-  function format(type: LogType, msg: string, options: LogErrorOptions = {}) {
+  function preventOverflow(msg: string) {
+    if (msg.length > MAX_LOG_CHAR) {
+      const shorten = msg.slice(0, MAX_LOG_CHAR)
+      const shortenLines = shorten.match(splitRE)?.length || 0
+      const lines = msg.match(splitRE)?.length || 0 - shortenLines
```

Comment:
Could we do `msg.slice(MAX_LOG_CHAR).match(splitRE)?.length` instead so we can get the lines with one regex call?

Reply:
`lines` is "lines of the original log" - "lines of shortened log".

In my view, 2 regex call is needed.

Could you explain a little bit more on reducing the regex call?

Reply:
I mean instead of "lines of the original log" - "lines of shortened log", what about  "lines of the (original log - shortened log)"? We can calculate the deleted loggings first then count the lines.

Reply:
Ah, understood. Thank you!